zerver/views/messages.py

```diff
@@ -863,6 +866,40 @@ def json_update_message(request, user_profile):
     return update_message_backend(request, user_profile)
 
 @has_request_variables
+def highlight_message_differences(message_history, sender_email):
+    # type: (List[Dict[str, Any]], str) -> List[Dict[str, Any]]
+    for edited_msg, prev_msg in zip(message_history, message_history[1:]):
+        # Iterate over message edit history
+        rendered_content = highlight_html_differences(prev_msg['prev_content'], \
+                                                      edited_msg['prev_content'])
+        edited_msg_index = message_history.index(edited_msg)
+        message_history[edited_msg_index]['prev_rendered_content'] = rendered_content
```

Comment:
Ahh, the raw vs. rendered issue is here -- you're computing the differences between the unrendered content and then sending it as though it's rendered content.  I think probably the right API is to highlight HTML differences in both the rendered (HTML) and unrendered (markdown input) content on the backend, and then just let the frontend decide what to display (I kinda think we'll eventually end up adding a tab switcher to show either, but it's not needed for v1).